src/systems/user_commands/UserCommands.cc

```diff
@@ -725,9 +725,28 @@ bool CreateCommand::Execute()
     }
     case msgs::EntityFactory::kCloneName:
     {
-      // TODO(louise) Implement clone
-      ignerr << "Cloning an entity is not yet supported." << std::endl;
-      return false;
+      auto entityToClone = this->iface->ecm->EntityByComponents(
+          components::Name(createMsg->clone_name()));
+      auto parentComp =
+        this->iface->ecm->Component<components::ParentEntity>(entityToClone);
```

Comment:
For now, it's best we shortcut here if `!parentComp || parentComp->Data() != worldEntity`, because we don't have good support for creating non-top-level entities yet. i.e. we don't want to let users clone a joint just yet.

Reply:
Sounds good. I made the change in https://github.com/ignitionrobotics/ign-gazebo/pull/959/commits/ba5f116ae6dbcc4602a15853cec2dae1d9d7aec6 and added a `TODO` note about adding future support for creating non-top level entities